pil-analyzer/src/evaluator.rs

```diff
@@ -749,7 +749,12 @@ impl<'a, 'b, T: FieldElement, S: SymbolLookup<'a, T>> Evaluator<'a, 'b, T, S> {
             Expression::BlockExpression(_, BlockExpression { statements, expr }) => {
                 self.op_stack
                     .push(Operation::TruncateLocals(self.local_vars.len()));
-                self.op_stack.push(Operation::Expand(expr));
+                match expr {
+                    Some(expr) => {
+                        self.op_stack.push(Operation::Expand(expr));
+                    }
+                    None => {}
```

Comment:
Do you have tests covering this line? Shouldn't it at least push an empty tuple or something?

Reply:
My guess here was that it made no difference to expand an empty expression, hence the empty None case. Also, there is the problem that creating an expression at this point is complex because of the lifetimes.


I'll see what I can do to solve it, thanks for your comment.

Reply:
This test should cover the case you mention, or am I wrong?

https://github.com/powdr-labs/powdr/blob/524d579544eb4d3f2370a5d8af0cb1011ded5546/pil-analyzer/tests/condenser.rs#L208

Reply:
I don't know. Can you put a panic inside there to see if it is hit?

Also can you add a test for the following (also just curious what is happening and if we should disallow it):
```
let x = {};
```

Reply:
Oh or maybe also something a bit more complicated:
```
let f = |_| 7;
let g = f({});
```

Reply:
Fixed. To avoid the lifetime issue, what I did was to change the None cases to the empty tuple directly in the statement processing, so at this point, the None case should not be reachable (I added some tests, including this one you mention above).

Reply:
Are you sure this is a good solution? I don't really like modifying the AST in such a way. You don't need to deal with the lifetime of expressions here. For an example, see the ::Number case. I think the following should do it:

```
 self.value_stack.push(Value::Tuple(vec![]))
```